test/org/sagebionetworks/bridge/services/ScheduledActivityServiceMockTest.java

```diff
@@ -600,11 +603,81 @@ public void taskNotStartedIsUpdatedV4() {
                 .withSurvey("my-survey", "my-survey-guid", new DateTime(1234)).build();
         List<ScheduledActivity> db = createNewActivities("CCC"+TIME_PORTION);
         db.get(0).setActivity(oldActivity);
-        when(activityDao.getActivities(eq(TIME_ZONE), any())).thenReturn(db);
+        
+        ForwardCursorPagedResourceList<ScheduledActivity> page = new ForwardCursorPagedResourceList<>(db, null);
+        when(activityDao.getActivityHistoryV2(any(), any(), any(), any(), any(), anyInt())).thenReturn(page);
         
         List<ScheduledActivity> returnedActivities = service.getScheduledActivitiesV4(createScheduleContext(NOW).build());
         assertEquals(1, returnedActivities.size());
         assertEquals(5678, returnedActivities.get(0).getActivity().getSurvey().getCreatedOn().getMillis());
+        
+        verify(activityDao).getActivityHistoryV2(any(), any(), any(), any(), any(), anyInt());
+    }
+    
+    @Test
+    public void taskNotStartedWithClientDataNotUpdated() {
+        // Activity in DDB has survey reference pointing to createdOn 1234. Newly created activity has survey reference
+        // pointing to createdOn 5678. Activity in DDB hasn't been started. We should update the activity in DDB.
```

Comment:
Minor: Update comment to reflect the new reality. (Here and in the next test.)

Reply:
Updated